lib/rucio/cli/replica.py

```diff
@@ -60,13 +65,18 @@ def list_(ctx, dids, protocols, all_states, pfns, domain, link, missing, metalin
 
 @replica_list.command("dataset")
 @click.argument("dids", nargs=-1)
+@click.option("--rse", default=None, help="RSE name to use a filter")
 @click.option("--deep", default=False, is_flag=True, help="Make a deep check, checking the contents of datasets in datasets")
 @click.option("--csv", help="Write output to comma separated values", is_flag=True, default=False)
 @click.pass_context
-def list_dataset(ctx, dids, deep, csv):
-    """List dataset replicas"""
-    args = Arguments({"no_pager": ctx.obj.no_pager, "dids": dids, "deep": deep, "csv": csv})
-    list_dataset_replicas(args, ctx.obj.client, ctx.obj.logger, ctx.obj.console, ctx.obj.spinner)
+def list_dataset(ctx, dids: Optional["Sequence[str]"], rse: Optional[str], deep: bool, csv: bool):
+    """List dataset replicas, or view all datasets at a RSE"""
```

Comment:
nit: I think "all datasets on a RSE" might be more grammatically correct

Reply:
(and I would also argue that it should be "an RSE", since the spelling of RSE begins with a vowel sound: https://blog.apastyle.org/apastyle/2012/04/using-a-or-an-with-acronyms-and-abbreviations.html, but we would then enter the territories of extreme nitpicking)